lib/modules/datasource/maven/util.ts

```diff
@@ -312,25 +333,21 @@ export async function downloadMavenXml(
   const protocol = pkgUrl.protocol;
 
   if (protocol === 'http:' || protocol === 'https:') {
-    const res = await downloadHttpProtocol(http, pkgUrl);
-    const body = res?.body;
-    if (body) {
-      return {
-        xml: new XmlDocument(body),
-        isCacheable: !res.authorization,
-      };
-    }
+    const rawResult = await downloadHttpProtocol(http, pkgUrl);
+    const xmlResult = rawResult.transform(({ isCacheable, data }): MavenXml => {
+      const xml = new XmlDocument(data);
+      return { isCacheable, xml };
+    });
+    return xmlResult.unwrapOr({});
```

Comment:
does it no longer throw if invalid XML?

Reply:
can you please log a message for invalid XML. eg if a user configured something wrong for the registry url?

Reply:
I'll have better handling of it it in further PR

Reply:
Could we live about a week without XML handling?